pkg/internal/infrastructure/terraform.go

```diff
@@ -342,3 +353,51 @@ func ComputeStatus(tf terraformer.Terraformer, config *api.InfrastructureConfig)
 
 	return status, nil
 }
+
+func fillDomainCounts(val map[string]interface{}, cluster *controller.Cluster, infra *extensionsv1alpha1.Infrastructure) error {
+	var (
+		faultDomainCount  *int32
+		updateDomainCount *int32
+	)
+
+	if infra.Status.ProviderStatus != nil {
+		infrastructureStatus, err := helper.InfrastructureStatusFromInfrastructure(infra)
+		if err != nil {
+			return err
+		}
+		nodesAvailabilitySet, err := helper.FindAvailabilitySetByPurpose(infrastructureStatus.AvailabilitySets, api.PurposeNodes)
+		if err != nil {
+			return err
+		}
+
+		if nodesAvailabilitySet != nil {
+			faultDomainCount = nodesAvailabilitySet.CountFaultDomains
+			updateDomainCount = nodesAvailabilitySet.CountUpdateDomains
```

Comment:
We can already return here, right? We have found the domain counts in the infra status. No need to check also the cloud profile.

```suggestion
			val["countFaultDomains"] = *nodesAvailabilitySet.CountFaultDomains
			val["countUpdateDomains"] = *nodesAvailabilitySet.CountUpdateDomains
			return nil
```

Reply:
Hm, I don't think so. When we introduce this change, we won't have these fields in the status during the first reconciliation.

Reply:
Ah right, yes for the case when we have already an infra status but the domain counts are not part of it.
Thanks for the explanation :)

Reply:
Nice catch. Probably a comment would help here - I was about to suggest the same like @dkistner.